src/pg/test/db/dbWrite.test.js

```diff
@@ -82,4 +104,66 @@ describe('postgres', () => {
       $put: true,
     });
   });
+
+  test('with $put, other table, (insert on conflict)', () => {
+    mockQuery.mockReturnValueOnce({
+      rowCount: 1,
+    });
+
+    const data = {
+      $key: { userId: 'userId_01' },
+      token: 'test',
+      $put: true,
+    };
+
+    store.write('googleSession', data);
+    const sqlQuery = sql`
+     INSERT INTO "googleSession" ( "token" , "version" )
+     VALUES ( ${
+       data.token
+     } , cast ( extract ( epoch from now ( ) ) as integer ) ) ON CONFLICT ( "userId" )
+     DO UPDATE SET ( "token" , "version" ) = (  ${
+       data.token
+     } , cast ( extract ( epoch from now ( ) ) as integer ) )
+     RETURNING ( to_jsonb ( "googleSession" ) || jsonb_build_object ( '$key' , ${`{"userId":"userId_01"}`}::jsonb , '$ref' , array[${`googleSession`} , "id"] , '$ver' , cast ( extract ( epoch from now ( ) ) as integer ) ) )
+    `;
+    expectSql(mockQuery.mock.calls[0][0], sqlQuery);
+  });
+
+  test('with $put, other table, (insert on conflict) v2', () => {
+    mockQuery.mockReturnValueOnce({
+      rowCount: 1,
+    });
+    const data = {
+      userId: 'userId_01',
+      token: 'test',
+      $put: true,
+    };
+
+    store.write(['googleSession', { userId: 'userId_01' }], data);
+    const sqlQuery = sql`
+     INSERT INTO "googleSession" ( "token" , "userId", "version" )
+     VALUES ( ${data.token} , ${
+      data.userId
+    }, cast ( extract ( epoch from now ( ) ) as integer ) ) ON CONFLICT ( "userId" )
+     DO UPDATE SET ( "token" ,"userId", "version" ) = (  ${data.token}, ${
+      data.userId
+    }, cast ( extract ( epoch from now ( ) ) as integer ) )
+     RETURNING ( to_jsonb ( "googleSession" ) || jsonb_build_object ( '$key' , ${`{"userId":"userId_01"}`}::jsonb , '$ref' , array[${`googleSession`} , "id"] , '$ver' , cast ( extract ( epoch from now ( ) ) as integer ) ) )
+    `;
+    expectSql(mockQuery.mock.calls[0][0], sqlQuery);
+  });
+
+  test.only('with $put, table with link, (insert on conflict)', () => {
```

Comment:
please check this test! Line 141 of Db.js should be `LinkResult` not `LinkChange`. Also `LinkChange` needs `query` to make it works.